### DISCUSSION TITLE
How to solve OCCLUSION in object tracking?

### DISCUSSION THREAD

**Discussion by RajashekarY** [STVIR/pysot#422] [General]
Hello everyone out  there we are looking for ideas to solve occlusion in object tracking anybody out there had ever had and issue like this or solved something near similar to this please share your VIEWS  regarding this aspect and SORRY if this is an irrelevant question being posted here please direct me in right way @StrangerZhang  @zyc-ai @lb1100 !!

refer:: https://photos.app.goo.gl/5xZtEUpnDz95xSyY8

Thanks

**Comment by ZhiyuanChen** [comment]
Actually based on the case in the provided link, I believe it is not an issue of occlusion. This is the case which the tracking object of face changed to the back of head. Siamese network prefer to track object, and the target object in DaSiamRPN. So I would try some data aug to make it identifies face and back of head be the same class. But your case is extremely difficult for Siamese network, I believe it would be a CVPR idea if you solve it.

**Comment by RajashekarY** [comment]
Hi @zyc-ai can you see the other video added just now to the same link above this is other example where I can show you what happened was false identification due to occlusion...
![occulsion_basket](https://user-images.githubusercontent.com/15843662/92388432-e6250280-f134-11ea-9047-b0fdd8499542.gif)

**Comment by MuruganR96** [comment]
@RajashekarY sir, I am also facing this **occlusion** issue. when i was tracking a object, if some other object is enter, tracking bounding box shifted to other object. 

 @zyc-ai @StrangerZhang @lb1100 Please give me suggstions, how to solve **occlusion**? what kind of changes/methods i need to do? 

Thanks,
Murugan Rajenthiran

**Comment by ZhiyuanChen** [comment]
> @RajashekarY sir, I am also facing this **occlusion** issue. when i was tracking a object, if some other object is enter, tracking bounding box shifted to other object.
> 
> @zyc-ai @StrangerZhang @lb1100 Please give me suggstions, how to solve **occlusion**? what kind of changes/methods i need to do?
> 
> Thanks,
> Murugan Rajenthiran

Again, this would be a CVPR idea, and if any of us knows how to solve it, we would have published a paper about the process.

**Comment by MuruganR96** [comment]
@zyc-ai @StrangerZhang @lb1100 @RajashekarY , here my ideas is,

### idea 1: prepare datasets without occlusion
**data** : video frames
**label**:  bbox values of corresponding frames

so with the video frames and bbox label,  
if we prepare datasets like, where occlusion present, will ignore those frames bounding box label. 

### idea 2: object re-identification 
If we develop re-identification stronger then we can resolve object mis-tracking after occlusion

Can we solve occlusion, when we implementing these ideas? which is better idea can go further?
@zyc-ai @StrangerZhang @lb1100 Please guide me in right way.

Thanks,
Murugan Rajenthiran

**Reply by ZhiyuanChen** [reply]
Sorry for this late reply. 

> Can we solve occlusion, when we implementing these ideas? 

Remove all scenes with occlusion will definitely "solve" the problem, as long as you can ensure such scene will not appear in the test set or where it is deploied to. 

It could solve the problem in some means. However, it could be a bit hard, especially when the occluded objects look identical. 

> which is better idea can go further?

First is better for engineering, second is better for academic.
